gato/gatos/ReiGato.py

```diff
@@ -0,0 +1,123 @@
+    ANIMATIONS = "reigato"
+    DISPLAY_NAME = "Mint Chocolate Parfait"
+    RARITY = 5
+    VALUES_TO_SAVE = ABaseGato.VALUES_TO_SAVE + [
+        "mint_gain_cooldown",
+    ]
+
+    REI_STATS_UP_EVENT_TYPE = "rei_stats_up"
+    REI_MINT_GAIN_EVENT_TYPE = "rei_mint_gain"
+    REI_MINT_RESTORE_EVENT_TYPE = "rei_mint_restore"
+    EVENT_DESCRIPTIONS = ABaseGato.EVENT_DESCRIPTIONS | {
+        REI_STATS_UP_EVENT_TYPE: "increased all allies' energy and mood! x{count}",
+        REI_MINT_GAIN_EVENT_TYPE: "gained a stack of mint! x{count}",
+        REI_MINT_RESTORE_EVENT_TYPE: "consumed a stack of mint and restored mood and hunger of an ally! x{count}",
+    }
+
+    # Mutable state variables used for this gato
+    # Cooldown till a mint stack can be gained again
+    mint_gain_cooldown: int = 0
+    # Current number of mint stacks
+    mint_stacks: int = 0
+
+    def mint(self, team, seconds):
+        # Tick down mint gain cd if needed
+        if self.mint_gain_cooldown > 0:
+            self.mint_gain_cooldown = max(self.mint_gain_cooldown - seconds, 0)
+
+        # Add a stack if not in cooldown or at max
+        if self.mint_gain_cooldown <= 0 and self.mint_stacks < 3:
+            self.mint_stacks += 1
+            self.set_mint_gain_cooldown()
+            # Send event
+            self._events.append({self.REI_MINT_GAIN_EVENT_TYPE: None})
+
+        # Check ally mood and hunger level and recover mood and energy as necessary if mint stacks available
+        gatos = iter(team)
+        while self.mint_stacks > 0 and (g := next(gatos, None)) is not None:
+            mood_lost = g.max_mood - g.mood
+            hunger_lost = g.max_hunger - g.hunger
+            if mood_lost + hunger_lost >= self.MINT_RESTORE_THRESHOLD:
+                # Calculate percentage increase
+                restore_percentage = self.MINT_BASE_RESTORE_PERCENTAGE + \
+                    max(5, self.eidolon) * \
+                    self.ADDITIONAL_RESTORE_PERCENTAGE_PER_EIDOLON
+                mood_restore_amount = g.max_mood * restore_percentage / 100
+                energy_restore_amount = g.max_energy * restore_percentage / 100
+                # Calculate additional flat increase if enhanced mint
+                if self.is_enhanced_mint():
+                    mood_restore_amount += self.ENCHANCED_MINT_ADDITIONAL_RESTORE_AMOUNT
+                    energy_restore_amount += self.ENCHANCED_MINT_ADDITIONAL_RESTORE_AMOUNT
+                # Apply restore
+                g.add_mood(mood_restore_amount)
+                g.add_energy(energy_restore_amount)
+                self.mint_stacks -= 1
+                self._events.append({self.REI_MINT_RESTORE_EVENT_TYPE: None})
+
+    def deploy(self, team: list["ABaseGato"]):
+        """Increase mood and energy for the team on deploy"""
+
+        # Run base class deploy
+        super().deploy(team)
+
+        # Reset mutable states
+        self.set_mint_gain_cooldown()
+        self.mint_stacks = 0
```

Comment:
Not actually sure if I should be doing this. 

Main thing that made me want to add this is to have an initial cooldown on the first deploy. I guess I can set it directly at the top